pkgs/os-specific/linux/kernel-headers/4.4.nix

```diff
@@ -1,43 +1,35 @@
-{ stdenv, fetchurl, perl, cross ? null }:
+{ stdenvNoCC, lib, buildPackages
+, buildPlatform, hostPlatform
+, fetchurl, perl
+}:
 
-assert cross == null -> stdenv.isLinux;
+assert hostPlatform.isLinux;
 
 let
-
   version = "4.4.10";
-
-  kernelHeadersBaseConfig =
-    if cross == null
-    then stdenv.platform.kernelHeadersBaseConfig
-    else cross.platform.kernelHeadersBaseConfig;
-
+  inherit (hostPlatform.platform) kernelHeadersBaseConfig;
 in
 
-stdenv.mkDerivation {
+stdenvNoCC.mkDerivation {
   name = "linux-headers-${version}";
 
   src = fetchurl {
     url = "mirror://kernel/linux/kernel/v4.x/linux-${version}.tar.xz";
     sha256 = "1kpjvvd9q9wwr3314q5ymvxii4dv2d27295bzly225wlc552xhja";
   };
 
-  targetConfig = if cross != null then cross.config else null;
+  targetConfig = if hostPlatform != buildPlatform then hostPlatform.config else null;
 
-  platform =
-    if cross != null then cross.platform.kernelArch else
-    if stdenv.system == "i686-linux" then "i386" else
-    if stdenv.system == "x86_64-linux" then "x86_64" else
-    if stdenv.system == "powerpc-linux" then "powerpc" else
-    if stdenv.isArm then "arm" else
-    if stdenv.platform ? kernelArch then stdenv.platform.kernelArch else
-    abort "don't know what the kernel include directory is called for this platform";
+  platform = hostPlatform.platform.kernelArch or (
```

Comment:
I think this fallback code (when `kernelArch` is not defined) should really go away.

Reply:
`kernelArch` is always defined because the kernel derivation uses it unconditionally. And it must not be something else than `kernelArch` because the name we need to pass here is not something standard but an implementation detail of the Linux kernel.